### PR TITLE
Upstream changes for v0.6.0 release

### PR DESCRIPTION
**Pull request by nv-pranjald**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,43 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.6.0] - 2024-05-06
+
+### Added
+- Simple RAG Example build a basic Retrieval-Augmented Generation (RAG) system.
```
Comment (shubhadeepd) [inline CHANGELOG.md:9]: `A simple rag example template showcasing how to build an example from scratch.`
And point to the documentation here.
Add this point to the bottom of list

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,43 @@ All notable changes to this project will be documented in this file.
...
+- Easily switch from [API Catalog](https://build.nvidia.com/explore/discover) models to locally deployed models using NVIDIA's enterprise solution (NIM) for enhanced production use.
```
Comment (shubhadeepd) [inline CHANGELOG.md:10]: Point to NIM documentation

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,43 @@ All notable changes to this project will be documented in this file.
...
+- New APIs
+  - `/health` - Provides a health check for the chain server.
+  - `/delete` - Allows deletion of ingested documents.
```
Comment (shubhadeepd) [inline CHANGELOG.md:13]: list and delete were added in previous release!

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,43 @@ All notable changes to this project will be documented in this file.
...
+  - `/list` - Lists all ingested documents.
+- Evaluator support to quantify performance of RAG pipeline
```
Comment (shubhadeepd) [inline CHANGELOG.md:15]: `A new evaluation application to quantify accuracy of different provided RAG examples.`
Point to the directory here.

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,43 @@ All notable changes to this project will be documented in this file.
...
+- Added more [experimental examples](./experimental/README.md) showcasing new usecases.
+  - [NVIDIA Product Advisor](./experimental/product-advisor/)
```
Comment (shubhadeepd) [inline CHANGELOG.md:17]: Remove this

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,6 +3,43 @@ All notable changes to this project will be documented in this file.
...
+- New Notebooks
+  - Added [Chat with NVIDIA financial data](./notebooks/12_Chat_wtih_nvidia_financial_reports.ipynb) notebook.
+  - Added notebook showcasing [langraph agent handling](./notebooks/11_LangGraph_HandlingAgent_IntermediateSteps.ipynb)
```
Comment (shubhadeepd) [inline CHANGELOG.md:20]: Wrong spelling!
`langGraph`

File: CHANGELOG.md [hunk 7]
```diff
@@ -3,6 +3,43 @@ All notable changes to this project will be documented in this file.
...
+  - Using `UAE-Large-V1` for text splitter
+- Ingested data now persists across multiple sessions.
+- Updated langchain-nvidia-endpoints to version 0.0.11, enabling support for models like llama3.
+- Filename-based validation ensures only supported files (txt, md, pdf) can be processed.
+- The default output token length in the UI has been increased from 250 to 1024 for more comprehensive responses.
+- Updated version of llama-index, pymilvus.
+- Updated pgvector container to `pgvector/pgvector:pg16`
+- Replace `nv-ai-foundation` embedding engine name with `nvidia-ai-endpoints`
+- LLM Model Updates
+  - [Multiturn Chatbot](./RetrievalAugmentedGeneration/examples/multi_turn_rag/) now uses `ai-mixtral-8x7b-instruct` model response generation.
```
Comment (shubhadeepd) [inline CHANGELOG.md:39]: `.. model FOR response generation`

File: CHANGELOG.md [hunk 8]
```diff
@@ -3,6 +3,43 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.6.0] - 2024-05-06
+
+### Added
```
Comment (shubhadeepd) [inline CHANGELOG.md:8]: Add observability support for langchain based examples

File: experimental/README.md [hunk 9]
```diff
@@ -39,6 +39,10 @@ Experimental examples are sample code and deployments for RAG pipelines that are
 
   This example is a sample demonstration on how Large Language Models (LLMs) could be used to synthetically generate training data, which can then be used to adapt retriever models.
 
+* [NVIDIA Retail Product Advisor AI Workflow](./product-advisor)
```
Comment (shubhadeepd) [inline experimental/README.md:42]: Remove this

**Review by shubhadeepd — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (nv-pranjald) [reply CHANGELOG.md]: Thanks updated.

File: CHANGELOG.md (hunk 2 above)
Comment (nv-pranjald) [reply CHANGELOG.md]: added ref for nim document

File: CHANGELOG.md (hunk 3 above)
Comment (nv-pranjald) [reply CHANGELOG.md]: removed list and delete endpoint

File: CHANGELOG.md (hunk 4 above)
Comment (nv-pranjald) [reply CHANGELOG.md]: Replaced with Containerized [evaluation application](./tools/evaluation/) for RAG pipeline accuracy measurement as suggested by Sumit in other comment

File: CHANGELOG.md (hunk 5 above)
Comment (nv-pranjald) [reply CHANGELOG.md]: Removed it, Thanks.

File: CHANGELOG.md (hunk 6 above)
Comment (nv-pranjald) [reply CHANGELOG.md]: Fixed, Thanks.

File: CHANGELOG.md (hunk 7 above)
Comment (nv-pranjald) [reply CHANGELOG.md]: updated, thanks.

File: CHANGELOG.md (hunk 8 above)
Comment (nv-pranjald) [reply CHANGELOG.md:8]: Added it thanks.

File: experimental/README.md (hunk 9 above)
Comment (nv-pranjald) [reply experimental/README.md:42]: Removed it.

File: CHANGELOG.md [hunk 10]
```diff
@@ -3,6 +3,40 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.6.0] - 2024-05-07
+
+### Added
+- Easily switch from [API Catalog](https://build.nvidia.com/explore/discover) models to locally deployed models using NVIDIA's enterprise solution [(NIM)](https://docs.nvidia.com/ai-enterprise/#nims) for enhanced production use.
```
[the remark this answers is not in the document]
Comment (nv-pranjald) [reply CHANGELOG.md]: Updated, thanks.

File: CHANGELOG.md [hunk 11]
```diff
@@ -3,6 +3,40 @@ All notable changes to this project will be documented in this file.
...
+- New APIs
```
[the remark this answers is not in the document]
Comment (nv-pranjald) [reply CHANGELOG.md]: Updated, thanks.

File: CHANGELOG.md [hunk 12]
```diff
@@ -3,6 +3,40 @@ All notable changes to this project will be documented in this file.
...
+  - Added notebook showcasing [langgraph agent handling](./notebooks/11_LangGraph_HandlingAgent_IntermediateSteps.ipynb)
+- A [simple rag example template](https://nvidia.github.io/GenerativeAIExamples/latest/simple-examples.html) showcasing how to build an example from scratch.
+
+### Changed
+- Renamed example `csv_rag` to [structured_data_rag](./RetrievalAugmentedGeneration/examples/structured_data_rag/)
+- Model Engine name update
+  - `nv-ai-foundation` and `nv-api-catalog` llm engine are renamed to `nvidia-ai-endpoints`
+  - `nv-ai-foundation` embedding engine is renamed to `nvidia-ai-endpoints`
+- Embedding model update
+  - `developer_rag` example uses `UAE-Large-V1` for text splitter and embedding
```
[the remark this answers is not in the document]
Comment (nv-pranjald) [reply CHANGELOG.md]: Added hf link with `UAE-Large-V1` and removed text splitter mention.

File: CHANGELOG.md [hunk 13]
```diff
@@ -3,6 +3,40 @@ All notable changes to this project will be documented in this file.
...
+  - Using `ai-embed-qa-4` for api catalog examples instead of `nvolveqa_40k` as embedding model
+  - Using `UAE-Large-V1` for text splitter
+- Ingested data now persists across multiple sessions.
+- Updated langchain-nvidia-endpoints to version 0.0.11, enabling support for models like llama3.
+- Filename-based validation ensures only supported files (txt, md, pdf) can be processed.
```
[the remark this answers is not in the document]
Comment (nv-pranjald) [reply CHANGELOG.md]: Updated the info,

File: CHANGELOG.md [hunk 14]
```diff
@@ -3,6 +3,40 @@ All notable changes to this project will be documented in this file.
...
+- The default output token length in the UI has been increased from 250 to 1024 for more comprehensive responses.
+- Updated version of llama-index, pymilvus.
+- Updated pgvector container to `pgvector/pgvector:pg16`
+- Replace `nv-ai-foundation` embedding engine name with `nvidia-ai-endpoints`
```
[the remark this answers is not in the document]
Comment (nv-pranjald) [reply CHANGELOG.md]: Thanks, removed it.

File: deploy/compose/docker-compose-nim-ms.yaml [hunk 15]
```diff
@@ -0,0 +1,49 @@
+services:
+  nemollm-inference:
+    container_name: nemollm-inference-microservice
+    image: nvcr.io/ohlfw0olaadg/ea-participants/nim_llm:24.02.1
```
[the remark this answers is not in the document]
Comment (nv-pranjald) [reply deploy/compose/docker-compose-nim-ms.yaml]: Changed it back to `24.02`.

**Review by sumitkbh — APPROVED** [review]

**Review by shubhadeepd — APPROVED** [review]

**Pull request merged**